virtualbox: security update to 6.1.30; #3664

WhiredPlanck opened the pull request:
<!-- For description on topic creation and maintenance, please refer to [this Wiki article](https://wiki.aosc.io/developer/packaging/topic-based-maintenance-guideline/). -->

Topic Description
-----------------

Security update `virtualbox` to 6.1.30

Package(s) Affected
-------------------

`virtualbox`

Security Update?
----------------

<!-- If this topic contains security update(s), please uncomment "Yes,"
     mark with the `security` and `priority` labels, and make sure to mark your commits to relevant issue numbers.

     Please see GitHub's documentation on "Linking a pull request to an issue":

     https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue -->

Yes - Issue Number: #3664 
<!-- No -->

<!-- Please uncomment the "Build Order" section if your topic affects more than one package. -->

<!--
Build Order
-----------

Please describe in what order maintainers should build this pull request.
-->

Test Build(s) Done
------------------

**Primary Architectures**

<!-- Please remove any architecture to which this topic does not apply. -->

- [x] AMD64 `amd64`   
<!-- If this package involves a `+32` counterpart, please uncomment the line below. -->    
<!-- - [ ] 32-bit Optional Environment `optenv32` -->

<!-- If all package(s) affected by this topic is `noarch`, please use the stub below. -->
<!-- - [ ] Architecture-independent `noarch` -->


<!-- Maintainers should review file changes and, if the build script(s) affected complies with the
     [Package Styling Manual](https://wiki.aosc.io/developer/packaging/package-styling-manual/), please
     add `lgtm` label to this issue. -->

<!-- Maintainers and users may now test the packages in this topic and, once user/maintainer feedback indicates that the update(s) work as expected and find its quality satisfactory,
     another maintainer may now review this pull request and mark it as Approved. After which, the maintainer will build affected package(s) and upload them to the `stable` repository. -->

Update(s) Uploaded to Stable
----------------------------

**Primary Architectures**

<!-- Please remove any architecture to which this topic does not apply. -->

- [ ] AMD64 `amd64`   
<!-- If this package involves a `+32` counterpart, please uncomment the line below. -->
<!-- - [ ] 32-bit Optional Environment `optenv32` -->

<!-- If all package(s) affected by this topic is `noarch`, please use the stub below. -->
<!-- - [ ] Architecture-independent `noarch` -->

<!-- TODO: CI to auto-fill architectural progress. -->

WhiredPlanck committed:
virtualbox: security update to 6.1.28; #3664

WhiredPlanck committed:
vitualbox: switch to python 3

- Add patch to let app call python 3

WhiredPlanck committed:
virtualbox: try to polish build script

- Declare scripts adapted from Arch Linux
- Use absolute path everywhere
- Add missing verbosity
- Fix hard-coded version once and for all
- Cancel VBoxREM removal
- Use Python 3 to run needed script

WhiredPlanck committed:
virtualbox: fix typos and incorrect paths to file

WhiredPlanck committed:
virtualbox: update to 6.1.30

WhiredPlanck committed:
virtualbox: revert to fix hard-coded version in `postinst` and `prerm`

WhiredPlanck commented:
LGTM on AMD64

![图片](https://user-images.githubusercontent.com/47623588/143202032-bfa6a18b-113f-4c5a-b6a3-16111c508fae.png)

MingcongBai commented on extra-emulation/virtualbox/01-host/build:
```diff
@@ -1,90 +1,109 @@
+# Adapted from Arch Linux
 VER=$PKGVER
 
-./configure \
```
Missing abinfo.

MingcongBai commented on extra-emulation/virtualbox/01-host/build:
```diff
@@ -1,90 +1,109 @@
     --with-makeself=/usr/bin/echo
 
-source ./env.sh
+source "$SRCDIR"/env.sh
```
Missing abinfo.

MingcongBai commented on extra-emulation/virtualbox/01-host/build:
```diff
@@ -1,90 +1,109 @@
 
-# Forward port DKMS
-cat > "$PKGDIR"/usr/src/vboxhost-TEMPVER/dkms.conf << EOF
+abinfo "Forward port DKMS"
```
What?

MingcongBai commented on extra-emulation/virtualbox/01-host/build:
```diff
@@ -157,6 +176,5 @@ fi
-chmod +x "$PKGDIR"/usr/src/vboxhost-TEMPVER/do_dkms
-chmod 0755 "$PKGDIR"/usr/lib/virtualbox/VirtualBox
-
+chmod -v +x "$PKGDIR"/usr/src/vboxhost-$PKGVER/do_dkms
```
Missing abinfo.

MingcongBai commented on extra-emulation/virtualbox/02-guest/build:
```diff
@@ -1,24 +1,25 @@
 VER=$PKGVER
-source ./env.sh
```
Missing abinfo throughout.

WhiredPlanck commented:
And ... (remind myself):
- Don’t use abbreviations for topic names.
- One should modify `postinst`, `prerm` etc. via `sed` command in prepare to use the value of a variable (like `$PKGVER`).

WhiredPlanck committed:
virtualbox: add missing abinfo and polish existing abinfo

WhiredPlanck committed:
virtualbox: try to polish build scripts again

- Declare build script (guest) adapted from Arch Linux
- Continue to add missing abinfo and polish existing abinfo
- Remove unused or duplicated processes
- Use PKGVER variable directly in build scripts
